src/rgb/schema/state.rs

```diff
@@ -363,10 +363,15 @@ mod strict_encoding {
 
             macro_rules! write_min_max {
                 ($min:ident, $max:ident, $e:ident, $len:ident) => {
-                    let (min, max) = ($min.to_le_bytes().to_vec(), $max.to_le_bytes().to_vec());
+                    let (min, max) = (
+                        $min.to_le_bytes().to_vec(),
+                        $max.to_le_bytes().to_vec(),
+                    );
                     $e.write_all(&min)?;
                     $e.write_all(&max)?;
-                    $len += ::core::mem::size_of_val(&min) * 2
+                    // TODO: Investigate why ::core::mem::size_of_val(&min)
+                    //       is not giving correct length
+                    $len += min.len() * 2
```

Comment:
Of course this is a bug, I had to write `&$min`, not `&min` (so it computes length of a sliced type, which in contains length field in itself, not of a primitive type). Fixed